### DISCUSSION TITLE
Reorder animation on FlatList items using itemLayoutAnimation prop

### DISCUSSION THREAD

**Discussion by dawid-bytys** [software-mansion/react-native-reanimated#6322] [Q&A] [answered]
Hi, I'm having trouble creating the FlatList animation described here https://docs.swmansion.com/react-native-reanimated/docs/layout-animations/list-layout-animations. How to do it correctly so that there is an animation when “Reorder” button is pressed? Am I missing something here? If so, it's described a bit poorly on the site.

```ts
import Animated, { LinearTransition } from 'react-native-reanimated';

const transition = LinearTransition.springify().duration(500);

export function Screen() {
  const insets = useSafeAreaInsets();
  const [tasks, setTasks] = useState(['Task 1', 'Task 2', 'Task 3', 'Task 4', 'Task 5']);
  
  function reorderTasks() {
    setTasks((prevTasks) => [...prevTasks.reverse()]);
  }

  function addTask() {
    setTasks((prevTasks) => [...prevTasks, `Task ${prev.length + 1}`]);
  }

  function removeTask() {
    setTasks((prevTasks) => [...prevTasks.slice(0, prevTasks.length - 1)]);
  }

  const renderTask = useCallback(({ item }: ListRenderItemInfo<string>) => {
    return (
      <View
        style={{
          backgroundColor: 'green',
          height: 70,
          borderRadius: 20,
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Text>{item}</Text>
      </View>
    );
  }, []);

  return (
    <View style={[styles.wrapper, { paddingTop: insets.top }]}>
      <Animated.FlatList
        contentContainerStyle={styles.listContentWrapper}
        data={tasks}
        renderItem={renderTask}
        itemLayoutAnimation={transition}
      />
      <View style={{ flex: 1, flexDirection: 'row', alignItems: 'center', justifyContent: 'center' }}>
        <Button title="Reorder" onPress={reorderTasks} />
        <Button title="Add" onPress={addTask} />
        <Button title="Remove" onPress={removeTask} />
      </View>
    </View>
  );
}
```

https://github.com/user-attachments/assets/6331d07d-d9b2-4b3b-a250-b0934f94db7b

**Comment by MatiPl01** [comment, marked as answer]
Hey!
Thank you for noticing the issue. It turns out that you need to add the `keyExtractor` property to your `Animated.FlatList` if your list items have neither the `id` nor the `key` property.
In general, the default `keyExtractor` in react native looks for only these 2 properties in your list items. If neither of these exists, you should provide your own implementation for efficient list items updated. It looks like `itemLayoutAnimation` also needs to know item keys. In your case, the following implementation of `keyExtractor` will fix the issue:

```tsx
keyExtractor={(item) => item}
```

**Reply by MatiPl01** [reply]
We will add this information to docs. Thanks again for reporting the issue.

**Reply by MatiPl01** [reply]
Also, you can look at [this](https://github.com/software-mansion/react-native-reanimated/blob/main/apps/common-app/src/examples/LayoutAnimations/ListItemLayoutAnimation.tsx) example in our example app if it helps.
